### ISSUE TITLE
Exclude & Allowed Switches Not Behaving as Expected

### ISSUE THREAD

**Issue by 03k64serenity** [digininja/CeWL#91]
https://github.com/digininja/CeWL/blob/280bfe6f8f57a783cf447c47cfb38ad568177d00/cewl.rb#L814

When providing regex patterns in a file for the --exclude or in the command line argument for --allowed, cewl is not properly excluding and allowing offsite URLs based on the rules.

**Comment by digininja** [comment]
It only checks the path and not the domain looking at that line of code.
Are you expecting it to check the domain as well?

**Comment by 03k64serenity** [comment]
Right. I'd like to be able to limit the spider from crawling certain domains and allow it to crawl others based on a regex.

**Comment by digininja** [comment]
Not currently possible. You could easily tweak that line to check the
domain instead. I don't know the property off hand, but try domain instead
of path.

**Comment by 03k64serenity** [comment]
Sounds good. Will do. Hey, by the way...I had no idea you were the author of CeWL all these years seeing you on the interwebs, so I'm even more impressed and grateful for your contributions to the community.

**Comment by digininja** [comment]
Glad you like it.

If you get stuck, let me know, and I'll have a look for the right property
in the morning.

**Comment by spencer-dollahite** [comment]
https://github.com/spencer-dollahite/CeWL/blob/master/cewl.rb

This is the sort of approach/feature I'd like to see to have both an allowed and exclude pattern switch for the domain and path. I know the code here isn't perfect, but I think it is close enough for demo purposes. Thoughts?

**Comment by digininja** [comment]
I'll have a look as soon as I get chance.
